### DISCUSSION TITLE
Move `*Controller` query methods to models

### DISCUSSION THREAD

**Discussion by nimmolo** [MushroomObserver/mushroom-observer#1024] [Ideas]
I'm realizing there are a lot of query methods in some older controllers (as i'm reading through `NameController`). They're important for building views, but often are (or could be) separated from controller code. 

Since GraphQL does not have access to controller methods, but does have access to model methods, and these queries may be useful in the new interface, I'd like to start a refactor to move these to the models.

For an example, see `NameController` [`#needed_descriptions`](https://github.com/MushroomObserver/mushroom-observer/blob/6f1e5ccb2b1c330dbe729ab65ff27c4dfee9a8f4/app/controllers/name_controller.rb#L165) [`#show_name`](https://github.com/MushroomObserver/mushroom-observer/blob/6f1e5ccb2b1c330dbe729ab65ff27c4dfee9a8f4/app/controllers/name_controller.rb#L355)

What do you all think?

**Comment by pellaea** [comment]
I agree.  I was definitely a little fuzzy on what belonged in the
controller and what belonged in the model.
